### ISSUE TITLE
Ensure tests are reproducible

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#4357]
## Summary

Most of our database seeders are using `fakerphp/faker` to generate random data for testing. However, for specific test cases this is resulting tests that are no reproducible. 

## Example

For the `search-workflows.spec.js` we are searching for users requiring that they have a diploma and expecting at least one result. However, we are randomly setting that value for all users and there is a small chance (which is occuring) where no user seeded has a diploma so it causes a false positive.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/c65ca74177ce2ae578b92e2c3965bbb6d2362412/api/database/factories/UserFactory.php#L107

https://github.com/GCTC-NTGC/gc-digital-talent/blob/c65ca74177ce2ae578b92e2c3965bbb6d2362412/frontend/cypress/integration/talentsearch/search-workflows.spec.js#L30-L33

https://github.com/GCTC-NTGC/gc-digital-talent/blob/c65ca74177ce2ae578b92e2c3965bbb6d2362412/frontend/cypress/integration/talentsearch/search-workflows.spec.js#L50

## Solution

There are some options to solve this, in order of preference(?):

1. Improve our tests so that we cover edge cases less and make fewer assumptions about the data
    - However unlikely, it is a valid that we have no users with diplomas
2. Update our seeders to ensure we cover our test cases
    - Undesirable since it presents a biased view of how we **_expect_** the data to look rather than what it **_could_** look like

**Comment by vd1992** [comment]
I was just thinking about this, I noticed that yeah, tests were failing but it was due to seeding. 
Not a new occurrence though, seeder count has been bumped up a few times to circumvent this.

**Comment by esizer** [comment]
> seeder count has been bumped up a few times to circumvent this.

Yeah, this will make the chances that it fails lower but doesn't solve the actual issue where we are testing specific cases that may never exist. 

SEE:

> However unlikely, it is a valid that we have no users with diplomas

**Comment by patcon** [comment]
I think this issue is the one that was referred to in https://github.com/GCTC-NTGC/gc-digital-talent/pull/4623#issuecomment-1307456587: we need talent requests to have candidates created that actually exist for a specific filter. (more often than not, there are no candidates for the randomly generated candidate requests. could prob tweak the seeder to make less common filters much less likely to be enabled. OR could make them more procedural or hand-tuned.)

**Comment by petertgiles** [comment]
Right now talentsearch/search-workflows.cy.js is failing about 25% - 50% of the time due to not having seeded the right candidates to have a non-zero candidate count for the tests.

**Comment by patcon** [comment]
Third option: 
3. Don't use test data, but write helpers and commands to generate data needed for the test on the fly.

This is what I did when writing tests for Polis. Whenever I was testing a feature, the setup of the test would prepare the data I needed. It worked very well, and separated laravel seeded data (which is maybe best for human QA testing) from e2e tests.

I recall I did allow some commands to "shortcut" if data already existed (e.g., if a user I wanted to use was already created)
